etcdctl/command/cluster_health.go

```diff
@@ -15,72 +16,108 @@ import (
 
 func NewClusterHealthCommand() cli.Command {
 	return cli.Command{
-		Name:   "cluster-health",
-		Usage:  "check the health of the etcd cluster",
-		Flags:  []cli.Flag{},
+		Name:  "cluster-health",
+		Usage: "check the health of the etcd cluster",
+		Flags: []cli.Flag{
+			cli.BoolFlag{Name: "forever", Usage: "forever check the health every 10 second until CTRL+C"},
+		},
 		Action: handleClusterHealth,
 	}
 }
 
 func handleClusterHealth(c *cli.Context) {
+	forever := c.Bool("forever")
+	if forever {
+		sigch := make(chan os.Signal, 1)
+		signal.Notify(sigch, os.Interrupt)
+
+		go func() {
+			<-sigch
+			os.Exit(0)
+		}()
+	}
+
 	tr, err := getTransport(c)
 	if err != nil {
 		handleError(ExitServerError, err)
 	}
 
+	// TODO: update members when forever is set.
 	mi := mustNewMembersAPI(c)
 	ms, err := mi.List(context.TODO())
-	if err != nil {
-		handleError(ExitServerError, err)
-	}
-
-	cl := make([]string, 0)
-	for _, m := range ms {
-		cl = append(cl, m.ClientURLs...)
-	}
 
-	// check the /health endpoint of all members first
-
-	ep, rs0, err := getLeaderStatus(tr, cl)
-	if err != nil {
-		fmt.Println("cluster may be unhealthy: failed to connect", cl)
-		os.Exit(1)
-	}
+	for {
+		if err != nil {
+			fmt.Println("cluster may be unhealthy: failed to list members")
+			if forever {
+				time.Sleep(10 * time.Second)
+				continue
+			}
+			handleError(ExitServerError, err)
+		}
 
-	time.Sleep(time.Second)
+		cl := make([]string, 0)
+		for _, m := range ms {
+			cl = append(cl, m.ClientURLs...)
+		}
 
-	// are all the members makeing progress?
-	_, rs1, err := getLeaderStatus(tr, []string{ep})
-	if err != nil {
-		fmt.Println("cluster is unhealthy")
-		os.Exit(1)
-	}
+		// check the /health endpoint of all members first
```

Comment:
I think you left this here. I do not know why you left this here. I do not want to resolve this in this commit since it is irrelevant.